### PR TITLE
✨  Included dependency parsing for go 

### PR DESCRIPTION
**Pull request by naveensrinivasan**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
 feature

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

  * Included dependency parsing of go.mod files.
  * Parse vanity URL in go.mod to add dependencies
  * Updated k8s dependencies based on the vanity URL's.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

None

* **Other information**:
None

### CODE REVIEW

File: scripts/update/main.go [hunk 1]
```diff
@@ -61,12 +74,94 @@ func GetBazelDeps(repo RepositoryDepsURL) []Repository {
...
+	mods := &gomod{}
+	if err := json.Unmarshal(out.Bytes(), mods); err != nil {
+		log.Panic(err)
+	}
+	// Close the file
+	if err := tmpFile.Close(); err != nil {
+		log.Panic(err)
+	}
+
+	for _, match := range mods.Require {
```
Comment (dlorenc) [inline scripts/update/main.go:135]: I think this is actually a common misconception with go.mod files - they do not contain the full set of dependencies required to build an application. For that, you need to use `go list -m all`.

Comment (naveensrinivasan) [reply scripts/update/main.go:135]: I am guessing `go list -m all` gives transitive dependency. Do we need the transitive dependencies?

Comment (dlorenc) [reply scripts/update/main.go:135]: I don't think it's that simple but can't explain it perfectly myself. My understanding is that go.mod only contains dependencies that are not the "natural" versions that would be selected by MVS.

Comment (naveensrinivasan) [reply scripts/update/main.go:135]: So you think we should change the implementation to use the go list -m all instead of go.mod?

Comment (dlorenc) [reply scripts/update/main.go:135]: Yeah, I understand "go list -m all", but I don't understand go.mod :)

It's possible that go.mod is all we need if we only care about direct dependencies, but from a security standpoint transitive dependencies are just as important.

Comment (naveensrinivasan) [reply scripts/update/main.go:135]: I understand. I am going to try and see if we can go list -m all with just go.mod and go.sum.

Comment (naveensrinivasan) [reply scripts/update/main.go:135]: OK, my doubt was correct, can't get the `go list -m all` with just `go.sum` and `go.mod`. We would have to download the entire repository.

Comment (naveensrinivasan) [reply scripts/update/main.go:135]: I am going to change the implementation to download the repository and then run `go list -m all`

Comment (naveensrinivasan) [reply scripts/update/main.go]: Took care of it. Thanks

**Review by dlorenc — APPROVED** [review]

**Pull request merged**